data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -22,17 +47,155 @@
+        try {
+            if (!executorService.awaitTermination(
+                    calculateLongestThreadWaitingTime(), TimeUnit.MILLISECONDS)) {
+                LOG.info("Consumer threads are waiting for shutting down...");
+                executorService.shutdownNow();
+            }
+        } catch (InterruptedException e) {
+            if (e.getCause() instanceof InterruptedException) {
+                LOG.error("Interrupted during consumer shutdown, exiting uncleanly...");
+                executorService.shutdownNow();
+                Thread.currentThread().interrupt();
+            }
+        }
+        LOG.info("Consumer shutdown successfully...");
+    }
+
+    private long calculateLongestThreadWaitingTime() {
+        List<TopicsConfig> topicsList = sourceConfig.getTopics();
+        return topicsList.stream().
+                map(
+                        topics -> topics.getThreadWaitingTime().toSeconds()
+                ).
+                max(Comparator.comparingLong(time -> time)).
+                orElse(1L);
+    }
+
+    private Properties getConsumerProperties(TopicsConfig topicConfig) {
+        String schemaType = "";
+        Properties properties = new Properties();
+        properties.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG,
+                topicConfig.getAutoCommitInterval().toSecondsPart());
+        properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG,
+                topicConfig.getAutoOffsetReset());
+        properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, sourceConfig.getBootStrapServers());
+        properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG,
+                topicConfig.getAutoCommit());
+        properties.put(ConsumerConfig.GROUP_ID_CONFIG, topicConfig.getGroupId());
+        schemaType = getSchemaType(sourceConfig.getSchemaConfig().getRegistryURL(), topicConfig.getName(), sourceConfig.getSchemaConfig().getVersion());
+        if (schemaType.isEmpty()) {
+            schemaType = MessageFormat.PLAINTEXT.toString();
+        }
+        setPropertiesForSchemaType(properties, schemaType);
+        if (sourceConfig.getAuthType().equalsIgnoreCase(AuthenticationType.PLAINTEXT.toString())) {
+            setPropertiesForAuth(properties);
+        }
+        return properties;
+    }
 
-  private static final Logger LOG = LoggerFactory.getLogger(KafkaSource.class);
-  
-  @Override
-  public void start(Buffer<Record<Object>> buffer) {
-    LOG.info("Initiated the start() method...");
-  }
+    private void setPropertiesForSchemaType(Properties properties,final String schemaType) {
+        if (schemaType.equalsIgnoreCase(MessageFormat.JSON.toString())) {
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                    StringDeserializer.class);
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+        } else if (schemaType.equalsIgnoreCase(MessageFormat.PLAINTEXT.toString())) {
```

Comment:
This can be moved to default case if schemaType doesn't match anything then default to PLAINTEXT or we should throw exception if schema type is not one of expected type.

Reply:
If the schema type doesn't match anything, then by default plaintext would be selected.